### PR TITLE
TLS integration and tests added

### PR DESCRIPTION
**Pull request by blakehatch**
# Description

Added tests to TLS integration from @allada

Towards: #301

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Please also list any relevant details for your test configuration

## Checklist

- [x] Updated documentation if needed
- [x] Tests added/amended
- [x] `bazel test //...`  passes locally
- [x] PR is contained in a single commit, using `git amend` see some [docs](https://www.atlassian.com/git/tutorials/rewriting-history)

- - -
This change is [Reviewable](https://reviewable.io/reviews/TraceMachina/turbo-cache/316)

### CODE REVIEW

**Review by MarcusSorealheis — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/TraceMachina/turbo-cache/316)* status: 0 of 16 files reviewed, 1 unresolved discussion (waiting on @blakehatch)

___
*[`.DS_Store` line 0 at r1](https://reviewable.io/reviews/TraceMachina/turbo-cache/316#-NgyxxOY97wiZnosu1yR:-NgyxxOY97wiZnosu1yS:b5bchoy) ([raw file](https://github.com/TraceMachina/turbo-cache/blob/83fb9c91c39cd365b9c625c425d1c68fa7611e89/.DS_Store#L0)):*
Please add `.DS_Store` to `.gitignore`, maybe in a MacOS section, and remove it from the commit.

**Review by MarcusSorealheis — COMMENTED** [review]
> Reviewed 1 of 16 files at r1.
> *[Reviewable](https://reviewable.io/reviews/TraceMachina/turbo-cache/316)* status: 1 of 16 files reviewed, 2 unresolved discussions (waiting on @blakehatch)

___
*[`cas/cas_main.rs` line 413 at r1](https://reviewable.io/reviews/TraceMachina/turbo-cache/316#-NgyzAGt78qXU2BWEu7o:-NgyzAGt78qXU2BWEu7p:bln9a96) ([raw file](https://github.com/TraceMachina/turbo-cache/blob/83fb9c91c39cd365b9c625c425d1c68fa7611e89/cas/cas_main.rs#L413)):*
> ```XML
>                 return Err(Box::new(make_err!(
>                     Code::InvalidArgument,
>                     "Expected 1 key in file {}, found {} keys",
> ```

Does it make sense to test this behavior? My intuition wants us to test for 0 keys or more than 1 key in the `tls_config.key` file. 

I don't think it will ever happen, necessarily, but I want to watch for this regression. I also understand that there is cost with each added test. It does feel a bit like we are testing rustls.

**Review by allada — COMMENTED** [review]
> Reviewed 6 of 7 files at r3.
> *[Reviewable](https://reviewable.io/reviews/TraceMachina/turbo-cache/316)* status: 15 of 17 files reviewed, 12 unresolved discussions (waiting on @blakehatch and @MarcusSorealheis)

___
*[`cas/cas_main.rs` line 413 at r1](https://reviewable.io/reviews/TraceMachina/turbo-cache/316#-NgyzAGt78qXU2BWEu7o:-Nh2KAafACL3lEX-IrG7:bbjto7v) ([raw file](https://github.com/TraceMachina/turbo-cache/blob/83fb9c91c39cd365b9c625c425d1c68fa7611e89/cas/cas_main.rs#L413)):*
<details><summary><i>Previously, MarcusSorealheis (Marcus Eagan) wrote…</i></summary><blockquote>

Does it make sense to test this behavior? My intuition wants us to test for 0 keys or more than 1 key in the `tls_config.key` file. 

I don't think it will ever happen, necessarily, but I want to watch for this regression. I also understand that there is cost with each added test. It does feel a bit like we are testing rustls.
</blockquote></details>

Yeah, I dont think this  is something we should test. Besides, we simply only support exactly 1 key, in the future we may support more, so no need to test more than 1. If it was less than 1, it'd fail to connect with `curl`.
___
*[`cas/grpc_service/bytestream_server.rs` line 556 at r3](https://reviewable.io/reviews/TraceMachina/turbo-cache/316#-Nh2Kd0G8TK_7019uc6S:-Nh2Kd0G8TK_7019uc6T:bf1l61k) ([raw file](https://github.com/TraceMachina/turbo-cache/blob/2922baa690b614a2b0d4d69b4bc8a8e700bfc0e9/cas/grpc_service/bytestream_server.rs#L556)):*
> ```XML
>         resp
>     }
> }
> ```

nit:needs new line at end of file.

**Comment by blakehatch** [comment]
@allada Is there a way you want me to test with other LBs like on GCP or Azure? 

Also I'm not able to run the start_turbo_cache.sh script to start the terraform deployment example as I don't have the create_filesystem.sh file in my root directory. If you want to test out on your machine or if you can send me the file I do final verification the script is working.

**Review by allada — CHANGES_REQUESTED** [review]
> Reviewed 1 of 4 files at r5.
> *[Reviewable](https://reviewable.io/reviews/TraceMachina/turbo-cache/316)* status: 12 of 19 files reviewed, 11 unresolved discussions (waiting on @blakehatch and @MarcusSorealheis)

___
*[`integration_tests/simple_tls_test.sh` line 30 at r8](https://reviewable.io/reviews/TraceMachina/turbo-cache/316#-NhJu_JcCqPqoFKqZuhb:-NhJu_JcCqPqoFKqZuhc:b61fpo1) ([raw file](https://github.com/TraceMachina/turbo-cache/blob/6704e3f550f6e2d470a6272b206c38310aa171d3/integration_tests/simple_tls_test.sh#L30)):*
> ```Shell
> cd ../
> 
> terraform init
> ```

Can't we just check that TLS works in here simply by using something like:
```sh
RESULTS=$(curl --cacert ./local_rootca.crt https://127.0.0.1:50051/ok)
if [[ "$RESULTS" != "Ok" ]]; then
  echo "Expected curl to be able to get 'Ok' status via TLS"
  exit 1
fi
```

?

**Comment by blakehatch** [comment]
Test is now passing with a successful TLS handshake

**Review by allada — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/TraceMachina/turbo-cache/316)* status: 12 of 21 files reviewed, 17 unresolved discussions (waiting on @blakehatch and @MarcusSorealheis)

*[a discussion](https://reviewable.io/reviews/TraceMachina/turbo-cache/316#-NhMlY1N5Pb_4gjwvS0m:-NhMlY1N5Pb_4gjwvS0n:bwmflya) (no related file):*
Looks like you need to rebase too.

___
___
*[`deployment-examples/docker-compose/docker-compose.yml` line 28 at r9](https://reviewable.io/reviews/TraceMachina/turbo-cache/316#-NhMjOm20FNXWvrKEpo7:-NhMjOm20FNXWvrKEpo8:bgse65e) ([raw file](https://github.com/TraceMachina/turbo-cache/blob/f1390aadc10cfc90c9aef86f0885aae9df934366/deployment-examples/docker-compose/docker-compose.yml#L28)):*
> ```YAML
>     volumes:
>       - ${TURBO_CACHE_DIR:-~/.cache/turbo-cache}:/root/.cache/turbo-cache
>       - ./local_rootca.crt:/etc/ssl/certs/local_rootca.crt
> ```

There's no need to put these into the container's OS level directories.

Instead I suggest generating it and putting it in this folder somewhere called `example_rootca.crt` and `example_key.pem` and check in the files. Then just reference it as source files instead.
___
*[`deployment-examples/docker-compose/local-storage-cas.json` line 45 at r9](https://reviewable.io/reviews/TraceMachina/turbo-cache/316#-NhMkbLy4RtpBq-Jslxt:-NhMkbLy4RtpBq-Jslxu:b-vsa6fe) ([raw file](https://github.com/TraceMachina/turbo-cache/blob/f1390aadc10cfc90c9aef86f0885aae9df934366/deployment-examples/docker-compose/local-storage-cas.json#L45)):*
> ```JSON
>     "listen_address": "0.0.0.0:50051",
>     "tls": {
>       "cert_file": "/etc/ssl/certs/local_rootca.crt",
> ```

Putting this here will break a bunch of tests.

Instead just create a new `servers` entry that works on a different port that serves over TLS.
___
*[`integration_tests/simple_tls_test.sh` line 22 at r9](https://reviewable.io/reviews/TraceMachina/turbo-cache/316#-NhMkpOn6kQwBnAs9zvu:-NhMkpOn6kQwBnAs9zvv:bu8rd8q) ([raw file](https://github.com/TraceMachina/turbo-cache/blob/f1390aadc10cfc90c9aef86f0885aae9df934366/integration_tests/simple_tls_test.sh#L22)):*
> ```Shell
> echo "Results: "
> 
> RESULTS=$(curl --insecure -v --tlsv1.2 --cacert ./local_rootca.crt --key ./key.pem https://127.0.0.1:50051/ok 2>&1)
> ```

nit: I think you can use `-q` instead of suppressing all stderr.
___
*[`integration_tests/simple_tls_test.sh` line 22 at r9](https://reviewable.io/reviews/TraceMachina/turbo-cache/316#-NhMlBo414YkhIwaJsp-:-NhMlBo414YkhIwaJsp0:bjj8q9j) ([raw file](https://github.com/TraceMachina/turbo-cache/blob/f1390aadc10cfc90c9aef86f0885aae9df934366/integration_tests/simple_tls_test.sh#L22)):*
> ```Shell
> echo "Results: "
> 
> RESULTS=$(curl --insecure -v --tlsv1.2 --cacert ./local_rootca.crt --key ./key.pem https://127.0.0.1:50051/ok 2>&1)
> ```

nit: `/status`. It will always echo a string of `Ok`.
___
*[`integration_tests/simple_tls_test.sh` line 22 at r9](https://reviewable.io/reviews/TraceMachina/turbo-cache/316#-NhMlNZ2BBz7ikWoNP3u:-NhMlNZ2BBz7ikWoNP3v:b-li2hiz) ([raw file](https://github.com/TraceMachina/turbo-cache/blob/f1390aadc10cfc90c9aef86f0885aae9df934366/integration_tests/simple_tls_test.sh#L22)):*
> ```Shell
> echo "Results: "
> 
> RESULTS=$(curl --insecure -v --tlsv1.2 --cacert ./local_rootca.crt --key ./key.pem https://127.0.0.1:50051/ok 2>&1)
> ```

Also, remove `-v`, we don't need all the headers.
___
*[`integration_tests/simple_tls_test.sh` line 24 at r9](https://reviewable.io/reviews/TraceMachina/turbo-cache/316#-NhMkzPy7LNIJuTUjOpN:-NhMkzPy7LNIJuTUjOpO:b-f6gwvx) ([raw file](https://github.com/TraceMachina/turbo-cache/blob/f1390aadc10cfc90c9aef86f0885aae9df934366/integration_tests/simple_tls_test.sh#L24)):*
> ```Shell
> RESULTS=$(curl --insecure -v --tlsv1.2 --cacert ./local_rootca.crt --key ./key.pem https://127.0.0.1:50051/ok 2>&1)
> 
> echo "$RESULTS"
> ```

nit: `echo "Results from curl: $RESULTS"`

Otherwise the message has no context.

**Review by blakehatch — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/TraceMachina/turbo-cache/316)* status: 8 of 21 files reviewed, 10 unresolved discussions (waiting on @allada and @MarcusSorealheis)

*[a discussion](https://reviewable.io/reviews/TraceMachina/turbo-cache/316#-NhMlY1N5Pb_4gjwvS0m:-NhOGOYd2gbPsgiDxx15:b-896fix) (no related file):*
<details><summary><i>Previously, allada (Nathan (Blaise) Bruer) wrote…</i></summary><blockquote>

Looks like you need to rebase too.
</blockquote></details>

Done.

___
___
*[`cas/cas_main.rs` line 413 at r1](https://reviewable.io/reviews/TraceMachina/turbo-cache/316#-NgyzAGt78qXU2BWEu7o:-Nh2crdK6VGItTIyo-r0:b-896fix) ([raw file](https://github.com/TraceMachina/turbo-cache/blob/83fb9c91c39cd365b9c625c425d1c68fa7611e89/cas/cas_main.rs#L413)):*
<details><summary><i>Previously, allada (Nathan (Blaise) Bruer) wrote…</i></summary><blockquote>

Yeah, I dont think this  is something we should test. Besides, we simply only support exactly 1 key, in the future we may support more, so no need to test more than 1. If it was less than 1, it'd fail to connect with `curl`.
</blockquote></details>

Done.
___
*[`deployment-examples/docker-compose/docker-compose.yml` line 28 at r9](https://reviewable.io/reviews/TraceMachina/turbo-cache/316#-NhMjOm20FNXWvrKEpo7:-NhOG78QAIKgYVQGcGsi:b-7wgb48) ([raw file](https://github.com/TraceMachina/turbo-cache/blob/f1390aadc10cfc90c9aef86f0885aae9df934366/deployment-examples/docker-compose/docker-compose.yml#L28)):*
<details><summary><i>Previously, allada (Nathan (Blaise) Bruer) wrote…</i></summary><blockquote>

There's no need to put these into the container's OS level directories.

Instead I suggest generating it and putting it in this folder somewhere called `example_rootca.crt` and `example_key.pem` and check in the files. Then just reference it as source files instead.
</blockquote></details>

Removed two lines adding cert and key into container directory and now referencing them as source files in local-storage-cas.json in the new server tls config
___
*[`deployment-examples/docker-compose/local-storage-cas.json` line 45 at r9](https://reviewable.io/reviews/TraceMachina/turbo-cache/316#-NhMkbLy4RtpBq-Jslxt:-NhNv2Xe6JP2uqHfLyd8:b-w7k3nc) ([raw file](https://github.com/TraceMachina/turbo-cache/blob/f1390aadc10cfc90c9aef86f0885aae9df934366/deployment-examples/docker-compose/local-storage-cas.json#L45)):*
<details><summary><i>Previously, allada (Nathan (Blaise) Bruer) wrote…</i></summary><blockquote>

Putting this here will break a bunch of tests.

Instead just create a new `servers` entry that works on a different port that serves over TLS.
</blockquote></details>

Created new servers entry serving on port 50071
___
*[`integration_tests/simple_tls_test.sh` line 22 at r9](https://reviewable.io/reviews/TraceMachina/turbo-cache/316#-NhMkpOn6kQwBnAs9zvu:-NhNwp8J7CHHYD_nnbs9:blt2j8v) ([raw file](https://github.com/TraceMachina/turbo-cache/blob/f1390aadc10cfc90c9aef86f0885aae9df934366/integration_tests/simple_tls_test.sh#L22)):*
<details><summary><i>Previously, allada (Nathan (Blaise) Bruer) wrote…</i></summary><blockquote>

nit: I think you can use `-q` instead of suppressing all stderr.
</blockquote></details>

2>&1 at the end of the command is redirecting stderr to stdout not supressing it
___
*[`integration_tests/simple_tls_test.sh` line 22 at r9](https://reviewable.io/reviews/TraceMachina/turbo-cache/316#-NhMlNZ2BBz7ikWoNP3u:-NhNzupf2g_IcE9eyp5I:b-896fix) ([raw file](https://github.com/TraceMachina/turbo-cache/blob/f1390aadc10cfc90c9aef86f0885aae9df934366/integration_tests/simple_tls_test.sh#L22)):*
<details><summary><i>Previously, allada (Nathan (Blaise) Bruer) wrote…</i></summary><blockquote>

Also, remove `-v`, we don't need all the headers.
</blockquote></details>

Done.
___
*[`tools/cargo_shared.bzl` line 163 at r2](https://reviewable.io/reviews/TraceMachina/turbo-cache/316#-Ngz0Qq52jKShK1dnSHY:-Nh2g817CQDxl8y548pC:b-896fix) ([raw file](https://github.com/TraceMachina/turbo-cache/blob/b167b6487dec35de539c5bfe5d0c01e3400367f7/tools/cargo_shared.bzl#L163)):*
<details><summary><i>Previously, allada (Nathan (Blaise) Bruer) wrote…</i></summary><blockquote>

unused.
</blockquote></details>

Done.
___
*[`tools/cargo_shared.bzl` line 170 at r2](https://reviewable.io/reviews/TraceMachina/turbo-cache/316#-Ngz0AhSBP8r8bGDz5k7:-Nh2hxpB4t2Uzq-qg84X:b-896fix) ([raw file](https://github.com/TraceMachina/turbo-cache/blob/b167b6487dec35de539c5bfe5d0c01e3400367f7/tools/cargo_shared.bzl#L170)):*
<details><summary><i>Previously, allada (Nathan (Blaise) Bruer) wrote…</i></summary><blockquote>

unused.
</blockquote></details>

Done.
___
*[`tools/cargo_shared.bzl` line 173 at r2](https://reviewable.io/reviews/TraceMachina/turbo-cache/316#-Ngz07Eu7F1Bnvz1G8Es:-Nh2hyad4PqP8hNKOtpq:b-896fix) ([raw file](https://github.com/TraceMachina/turbo-cache/blob/b167b6487dec35de539c5bfe5d0c01e3400367f7/tools/cargo_shared.bzl#L173)):*
<details><summary><i>Previously, allada (Nathan (Blaise) Bruer) wrote…</i></summary><blockquote>

unused.
</blockquote></details>

Done.
___
*[`.DS_Store` line  at r1](https://reviewable.io/reviews/TraceMachina/turbo-cache/316#-NgyxxOY97wiZnosu1yR:-Nh2do2jBAFDp1ObC9H-:b-896fix) ([raw file](https://github.com/TraceMachina/turbo-cache/blob/83fb9c91c39cd365b9c625c425d1c68fa7611e89/.DS_Store#L)):*
<details><summary><i>Previously, MarcusSorealheis (Marcus Eagan) wrote…</i></summary><blockquote>

Please add `.DS_Store` to `.gitignore`, maybe in a MacOS section, and remove it from the commit.
</blockquote></details>

Done.

**Review by allada — COMMENTED** [review]
> Reviewed 1 of 16 files at r1, 5 of 9 files at r10, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/TraceMachina/turbo-cache/316)* status: 14 of 21 files reviewed, 11 unresolved discussions (waiting on @blakehatch and @MarcusSorealheis)

___
*[`deployment-examples/docker-compose/local-storage-cas.json` line 76 at r10](https://reviewable.io/reviews/TraceMachina/turbo-cache/316#-NhSuuLg5oSu_Tc8cdRl:-NhSuuLg5oSu_Tc8cdRm:b-bfjtb6) ([raw file](https://github.com/TraceMachina/turbo-cache/blob/93b934f636c85f911e52a8512ed51a0929bf3c53/deployment-examples/docker-compose/local-storage-cas.json#L76)):*
> ```JSON
>     "listen_address": "0.0.0.0:50071",
>     "tls": {
>       "cert_file": "/root/local_rootca.crt",
> ```

nit: We need to check these files into the repo and it needs to point to it. This file is used as an "example" that people need to be able to run inside and outside docker.

**Review by allada — APPROVED** [review]
:lgtm:

> Reviewed 3 of 9 files at r10, 2 of 2 files at r11, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/TraceMachina/turbo-cache/316#-:-Nhc4Eq7A4-_fdRL6Kin:bnfp4nl)* status: 19 of 23 files reviewed, 6 unresolved discussions (waiting on @blakehatch and @MarcusSorealheis)

___
*[`cas/cas_main.rs` line 431 at r11](https://reviewable.io/reviews/TraceMachina/turbo-cache/316#-Nhc3aL41L0VpsC8XXaz:-Nhc3aL41L0VpsC8XXb-:b-fdwpiw) ([raw file](https://github.com/TraceMachina/turbo-cache/blob/d2ae3023bafba7eaed4c0bbaaf61ac162b71b267/cas/cas_main.rs#L431)):*
> ```XML
>         let tcp_listener = TcpListener::bind(&socket_addr).await?;
>         let mut http = Http::new();
>         http.http2_keep_alive_interval(Duration::from_secs(10));
> ```

nit: Remove this, it's not part of this PR.
___
*[`cas/cas_main.rs` line 432 at r11](https://reviewable.io/reviews/TraceMachina/turbo-cache/316#-Nhc3ZgB6tnPIqLt33Cl:-Nhc3ZgB6tnPIqLt33Cm:b-fdwpiw) ([raw file](https://github.com/TraceMachina/turbo-cache/blob/d2ae3023bafba7eaed4c0bbaaf61ac162b71b267/cas/cas_main.rs#L432)):*
> ```XML
>         let mut http = Http::new();
>         http.http2_keep_alive_interval(Duration::from_secs(10));
>         http.http2_max_pending_accept_reset_streams(1000);
> ```

nit: Remove this, it's not part of this PR.
___
*[`deployment-examples/docker-compose/key.pem` line 1 at r11](https://reviewable.io/reviews/TraceMachina/turbo-cache/316#-Nhc2j888tvu66qv_ILR:-Nhc2j888tvu66qv_ILS:bx510e4) ([raw file](https://github.com/TraceMachina/turbo-cache/blob/d2ae3023bafba7eaed4c0bbaaf61ac162b71b267/deployment-examples/docker-compose/key.pem#L1)):*
> ```
> -----BEGIN PRIVATE KEY-----
> ```

nit: Can we rename these `example-do-not-use-in-prod-key.pem` and same with cert... We don't want people to use these keys, they are for demo only.
___
*[`deployment-examples/docker-compose/local-storage-cas.json` line 74 at r11](https://reviewable.io/reviews/TraceMachina/turbo-cache/316#-Nhc3vRd39SdE0YGQaps:-Nhc3vRd39SdE0YGQapt:b-eusrdo) ([raw file](https://github.com/TraceMachina/turbo-cache/blob/d2ae3023bafba7eaed4c0bbaaf61ac162b71b267/deployment-examples/docker-compose/local-storage-cas.json#L74)):*
> ```JSON
>   },
>   {
>     "listen_address": "0.0.0.0:50071",
> ```

nit: Put a line here saying:
`// Example of serving over TLS on port 50071.`

**Comment by MarcusSorealheis** [comment]
@blakehatch I will approve after you address the two open comments

**Comment by MarcusSorealheis** [comment]
There's also an awful lot of test failures.

**Review by blakehatch — COMMENTED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/TraceMachina/turbo-cache/316#-:-NhcnKkz0feN5AHxvqFb:bnfp4nl)* status: 16 of 25 files reviewed, 2 unresolved discussions (waiting on @allada and @MarcusSorealheis)

**Comment by MarcusSorealheis** [comment]
@aaronmondal Any idea why this build is failing?

**Comment by allada** [comment]
CI is failing because `.into()` needs to be removed. If you run `bazel test //...` it will tell you everything:
```
ERROR: /home/runner/work/turbo-cache/turbo-cache/cas/BUILD:3:12: Clippy cas/cas.clippy.ok failed: (Exit 1): process_wrapper failed: error executing command (from target //cas:cas) bazel-out/k8-opt-exec-2B5CBBC6/bin/external/rules_rust/util/process_wrapper/process_wrapper --arg-file bazel-out/k8-fastbuild/bin/external/crate_index__axum-0.6.19/axum_build_script.linksearchpaths ... (remaining 322 arguments skipped)

Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
error: useless conversion to the same type: `std::vec::Vec<u8>`
   --> cas/cas_main.rs:420:53
    |
420 |                 .with_single_cert(certs, PrivateKey(keys.into_iter().next().unwrap().into()))
    |                                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: consider removing `.into()`: `keys.into_iter().next().unwrap()`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#useless_conversion
    = note: `-D clippy::useless-conversion` implied by `-D warnings`

error: aborting due to previous error
```

**Review by aaronmondal — CHANGES_REQUESTED** [review]
> Reviewed 2 of 9 files at r12, 8 of 8 files at r13, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/TraceMachina/turbo-cache/316)* status: 19 of 23 files reviewed, 5 unresolved discussions (waiting on @allada, @blakehatch, and @MarcusSorealheis)

___
*[`-- commits` line 2 at r13](https://reviewable.io/reviews/TraceMachina/turbo-cache/316#-Nhr_GXNAgZ_5DT-bUU0):*
nit: You could shorten this to "Add TLS integration". No need to mention the tests :innocent: 
___
*[`cas/cas_main.rs` line 421 at r13](https://reviewable.io/reviews/TraceMachina/turbo-cache/316#-Nhr_waA2d-2yxydq1nY:-Nhr_waB4Laq69fVRbDx:b1rgja2) ([raw file](https://github.com/TraceMachina/turbo-cache/blob/a75c39b3eb6690de0700bd0964da7e678d4a084d/cas/cas_main.rs#L421)):*
> ```XML
>                 .with_safe_defaults()
>                 .with_no_client_auth()
>                 .with_single_cert(certs, PrivateKey(keys.into_iter().next().unwrap().into()))
> ```

The tests are failing because the bazel tests use clippy which isn't enabled for the Cargo Build. The error is this:

```
error: useless conversion to the same type: `std::vec::Vec<u8>`
   --> cas/cas_main.rs:421:53
    |
421 |                 .with_single_cert(certs, PrivateKey(keys.into_iter().next().unwrap().into()))
    |                                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: consider removing `.into()`: `keys.into_iter().next().unwrap()`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#useless_conversion
    = note: `-D clippy::useless-conversion` implied by `-D warnings`

error: aborting due to previous error
```
The fix is to remove the `into()` as the message says.

I'll try to get better integration for clippy for the Cargo build.
___
*[`integration_tests/simple_tls_test.sh` line 14 at r13](https://reviewable.io/reviews/TraceMachina/turbo-cache/316#-Nhrc1KN44Df9vjqQAC8:-Nhrc1KN44Df9vjqQAC9:b-bebo5p) ([raw file](https://github.com/TraceMachina/turbo-cache/blob/a75c39b3eb6690de0700bd0964da7e678d4a084d/integration_tests/simple_tls_test.sh#L14)):*
> ```Shell
> # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
> # See the License for the specific language governing permissions and
> # limitations under the License.
> ```

nit: Missing empty line.

**Review by MarcusSorealheis — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/TraceMachina/turbo-cache/316#-:-Nhs4rGjA2UZMOvfB9g0:bnfp4nl)* status: 18 of 23 files reviewed, 2 unresolved discussions (waiting on @aaronmondal, @allada, and @blakehatch)

**Review by MarcusSorealheis — COMMENTED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/TraceMachina/turbo-cache/316#-:-Nhs9mkQ3hW8iaUSSQ2f:bnfp4nl)* status: 18 of 23 files reviewed, 1 unresolved discussion (waiting on @aaronmondal and @allada)

**Review by aaronmondal — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r15, 1 of 1 files at r16, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/TraceMachina/turbo-cache/316#-:-NhscfOdCCVCH8tVzL7Q:bnfp4nl)* status: 19 of 23 files reviewed, all discussions resolved (waiting on @allada and @MarcusSorealheis)
